fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DbFedoraToOcflObjectIndex.java

```diff
@@ -265,7 +266,11 @@ private void upsert(final String transactionId, final FedoraId fedoraId, final S
         parameterSource.addValue("ocflId", ocflId);
         parameterSource.addValue("transactionId", transactionId);
         parameterSource.addValue("operation", operation);
-        jdbcTemplate.update(UPSERT_MAPPING_TX_MAP.get(dbPlatform), parameterSource);
+        try {
+            jdbcTemplate.update(UPSERT_MAPPING_TX_MAP.get(dbPlatform), parameterSource);
+        } catch (Exception e) {
+            throw new InvalidResourceIdentifierException("ID path too long");
```

Comment:
It is almost always a good idea to include the cause when throwing a new exception. You should do so here, unless there's a good reason not to.

Reply:
you mean change the message to be more descriptive?

Reply:
I mean include the cause in the new exception like:

```java
throw new InvalidResourceIdentifierException("message", e);
```